extensions/identity-hub/src/main/java/org/eclipse/dataspaceconnector/identityhub/processor/CollectionsWriteProcessor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.processor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.identityhub.models.MessageResponseObject;
+import org.eclipse.dataspaceconnector.identityhub.models.MessageStatus;
+import org.eclipse.dataspaceconnector.identityhub.store.IdentityHubStore;
+
+import java.io.IOException;
+
+import static org.eclipse.dataspaceconnector.identityhub.models.MessageResponseObject.MESSAGE_ID_VALUE;
+
+/**
+ * Processor of "CollectionsWrite" messages, in order to write objects into the {@link IdentityHubStore}.
+ */
+public class CollectionsWriteProcessor implements MessageProcessor {
+
+    private final IdentityHubStore identityHubStore;
+    private final ObjectMapper objectMapper;
+
+    public CollectionsWriteProcessor(IdentityHubStore identityHubStore, ObjectMapper objectMapper) {
+        this.identityHubStore = identityHubStore;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public MessageResponseObject process(byte[] data) {
+        Object hubObject;
+        try {
+            hubObject = objectMapper.readValue(data, Object.class);
```

Comment:
just as an idea:

Rather than storing plain java objects here maybe we can deserialize this into jackson ObjectNode/JsonNode. That can also be helpful for us to also parse the input and serialize e.g. logging easily if needed.

And then in-memory store can also store jackson ObjectNode/JsonNode directly. 

(No change for this PR but just an idea for other implementation stories)